src/session/get-access-token.ts

```diff
@@ -129,23 +129,24 @@ export default function accessTokenFactory(
       }
     }
 
-    // Check if the token has expired.
-    // There is an edge case where we might have some clock skew where our code assumes the token is still valid.
-    // Adding a skew of 1 minute to compensate.
-    if (!session.refreshToken && session.accessTokenExpiresAt * 1000 - 60000 < Date.now()) {
-      throw new AccessTokenError(
```

Comment:
Hi @kanalo-shrek - can you [fix the case](https://app.circleci.com/pipelines/github/auth0/nextjs-auth0/636/workflows/1320a98c-f2fa-4d55-8e7c-6aac69533ab0/jobs/696/parallel-runs/0/steps/0-107) where `The access token expired and a refresh token is not available`

and also add a test case for the scenario that you've introduced

Reply:
Ah I was relying on both the cases, but your comment on detecting the errors early makes sense, just for the sake of readability though do you feel like the SDK should declare variables as `const hasAccessTokenExpired = ...` etc, instead of repeating the same logic everywhere?

Reply:
Yes, thats true, although you don't need to touch that part of the code for your change (just add a couple of lines for the `!refreshToken && refresh` branch and a single test case for it)